infra/proposer_test.go

```diff
@@ -0,0 +1,31 @@
+package infra_test
+
+import (
+	"github.com/guoger/stupid/infra"
+	"github.com/hyperledger/fabric-sdk-go/pkg/fab/mocks"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Proposer", func() {
+
+	var addr string
+
+	BeforeEach(func() {
+		srv := &mocks.MockEndorserServer{}
+		addr = srv.Start("127.0.0.1:0")
+	})
+
+	Context("CreateProposer", func() {
+		It("successfully creates a proposer", func() {
+			Proposer := infra.CreateProposer(addr, &infra.Crypto{})
+			Expect(Proposer.Addr).To(Equal(addr))
```

Comment:
you still need to expect `Proposer` not be nil first, right?

Reply:
if Proposer is nil, Proposer.Addr will panic right?
so Proposer.Addr cover Proposer is not nil?